axelor-purchase/src/main/java/com/axelor/apps/purchase/service/PurchaseOrderWorkflowServiceImpl.java

```diff
@@ -62,7 +62,7 @@ public void draftPurchaseOrder(PurchaseOrder purchaseOrder) throws AxelorExcepti
   }
 
   @Override
-  @Transactional(rollbackOn = {Exception.class})
+  @Transactional(rollbackOn = {AxelorException.class})
```

Comment:
This was good before, this is a regression as it will not rollback on a NPE for example